gcloud/storage/_implicit_environ.py

```diff
@@ -81,29 +29,11 @@ class _DefaultsContainer(object):
     :param connection: Persistent implied connection from environment.
     """
 
-    @_lazy_property_deco
-    @staticmethod
-    def project():
-        """Return the implicit default project."""
-        return _determine_default_project()
-
-    def __init__(self, project=None, bucket=None, connection=None,
-                 implicit=False):
-        if project is not None or not implicit:
-            self.project = project
+    def __init__(self, bucket=None, connection=None):
```

Comment:
I agree. So is there reason to make a `gcloud._DefaultsContainer`? I'm torn because

- It reduces the startup cost of determining that value in an application using more than one package
- It makes maintenance (slightly) more difficult: A visual scan of each `_DefaultsContainer` won't reveal that a particular package is also using a default project. (Heading over to `__init__.py` will, but that requires knowing it's there in some sense.)